src/scene/particle-container/shared/ParticleContainer.ts

```diff
@@ -302,7 +302,12 @@ export class ParticleContainer extends ViewContainer implements Instruction
      */
     public removeParticles(beginIndex?: number, endIndex?: number)
     {
-        const children = this.particleChildren.splice(beginIndex, endIndex);
+        // Default beginIndex to 0 if not provided
+        const start = beginIndex ?? 0;
+        // Default endIndex to the length of particleChildren if not provided
+        const end = endIndex ?? this.particleChildren.length;
+        // Remove the correct range
+        const children = this.particleChildren.splice(start, end);
```

Comment:
Alternatively: 
```suggestion
        // Default beginIndex to 0 if not provided
        beginIndex ??= 0;
        // Default endIndex to the length of particleChildren if not provided
        endIndex ??= this.particleChildren.length;
        // Remove the correct range
        const children = this.particleChildren.splice(beginIndex, endIndex);
```

Reply:
This should be considered a breaking change because users may have removeParticles() calls in production. If their pixijs version isn't locked, their next release could exhibit unintentional behavior if they upgrade without thorough testing.

Reply:
tweaked this based on your feedback @bigtimebuddy . 
Thanks for youre insight @susiwen8! , whilst it is kind of a breaking change, it also is fixing incorrect behaviour. With that in mind, I think on this occasion its ok not to consider this breaking as the use case would be quite strange!